elastalert/elastalert.py

```diff
@@ -534,43 +540,57 @@ def get_hits_terms(self, rule, starttime, endtime, index, key, qk=None, size=Non
         return {endtime: buckets}
 
     def get_hits_aggregation(self, rule, starttime, endtime, index, query_key, term_size=None):
-        rule_filter = copy.copy(rule['filter'])
-        base_query = self.get_query(
-            rule_filter,
-            starttime,
-            endtime,
-            timestamp_field=rule['timestamp_field'],
-            sort=False,
-            to_ts_func=rule['dt_to_ts'],
-            five=rule['five']
-        )
-        if term_size is None:
-            term_size = rule.get('terms_size', 50)
-        query = self.get_aggregation_query(base_query, rule, query_key, term_size, rule['timestamp_field'])
+        agg_key = rule['metric_agg_type']+"("+rule['metric_agg_key']+")"
+        query = rule['filter'][0]['query_string']['query']
+        data, count = self.get_ch_data(rule, starttime, endtime, agg_key, query)
+        
+        if data is None:
+            return {}
+        payload = {rule['metric_agg_key']+"_"+rule['metric_agg_type']: {'value': data}}
+
+        self.num_hits += count
+        return {endtime: payload}
+
+    def get_error_rate(self, rule, starttime, endtime):
+        agg_key = rule['metric_agg_type']+"()"
+        query = rule['filter'][0]['query_string']['query']
+        total_data, total_count = self.get_ch_data(rule, starttime, endtime, agg_key, query)
+
+        if total_data is None:
+            return {}
+        
+        query = query+" AND "+rule['error_condition']
+        error_data, error_count = self.get_ch_data(rule, starttime, endtime, agg_key, query)
+
+        if error_data is None:
+            return {}
+
+        payload = {'error_count': error_data, 'total_count': total_data}
+        self.num_hits += int(total_count)
```

Comment:
why num_hits is required?

Reply:
num hits is mainly used in looping very large dataset in batches using scroll in ES.  Also it appears in the alert. Updating it to follow the convention followed across the code